executor/bin/prepare-executor.sh

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bash
+
+# USAGE: ./bin/prepare-executor.sh MODE TARGET_DIR
+# Builds the cook executor and then copies it to TARGET_DIR
+# Examples:
+#   ./bin/prepare-executor.sh docker /target/directory
+#   ./bin/prepare-executor.sh local /target/directory
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+EXECUTOR_DIR="$(dirname ${DIR})"
+MODE=${1}
```

Comment:
Rather than specifying a *mode* (`docker` vs `local`), I think it would be better to have the different versions based on your target *platform*. We could generate the `PLATFORM` string with something like `$(uname -s)-$(uname -i)`, which should yield `Linux-x86_64` or `Darwin-x86_64` for 64-bit Linux or OS X, respectively.

I think it's still fine to have `docker` vs `local` for the build script names (since that's describing where the build is done, not the target platform), but I'd prefer having the generated binary use the target platform in its name.

Reply:
I've undone changes that added the `-docker` suffix, the locally created executable will have the `-local` suffix.